packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -620,6 +626,23 @@ class DirectStripeRoutes {
    */
   async handleSubscriptionDeletedEvent(request, event) {
     const sub = /** @type {Subscription} */ (event.data.object);
+    await this.sendSubscriptionDeletedEmail(sub);
+    return this.updateCustomerAndSendStatus(request, event, sub, false);
+  }
+
+  /**
+   * Handle `invoice.payment_succeeded` Stripe wehbook events.
+   *
+   * @param {*} request
+   * @param {Event} event
+   */
+  async handleInvoicePaymentSucceededEvent(request, event) {
+    const invoice = /** @type {Invoice} */ (event.data.object);
+    await this.sendSubscriptionInvoiceEmail(invoice);
+    const sub = await this.stripeHelper.expandResource(
+      invoice.subscription,
+      'subscriptions'
+    );
```

Comment:
How come this doesn't fetch the customer (too bad we aren't dual keying our cache to stripe customer string) here? Also, it appears that when a payment is succeeded, we're about to send out a subscription state change telling our RP's to de-activate the subscription? (the 'false' in the call below). We should check the existing status of the customer before sending out subscription state changes, since someone that always pays on time isn't having their subscription state changed.

I believe there is a function we have that can refresh a customer given their uid/email, and send out a SQS message if something changed. That feels like a good fit here. (Assuming this function I remember exists)

Reply:
>How come this doesn't fetch the customer

Customer isn't needed here - `updateCustomerAndSendStatus` wants a subscription and fetches customer from cache itself.

> Also, it appears that when a payment is succeeded, we're about to send out a subscription state change telling our RP's to de-activate the subscription?

Switching this to `true`

> I believe there is a function we have that can refresh a customer given their uid/email, and send out a SQS message if something changed. That feels like a good fit here.

Isn't that what `updateCustomerAndSendStatus` is doing with `refreshCachedCustomer` and `profile.deleteCache`?

Reply:
> Customer isn't needed here - updateCustomerAndSendStatus wants a subscription and fetches customer itself from subscription info.

Right, I mean, we could possibly avoid 1 stripe call by having a function that can use a uid/email instead to fetch from our Redis customer cache and match the subscription id. It would need another function instead of `updateCustomerAndSendStatus`, so I guess it depends on how important it is to reduce the stripe API calls here.

> Isn't that what updateCustomerAndSendStatus is doing with refreshCachedCustomer and profile.deleteCache?

Nope, `updateCustomerAndSendStatus` will always send a SQS event. The `handleSubscriptionUpdatedEvent` function is using the prior attributes to determine if a subscription state change event is needed.

Upon further reflection walking my 🐕 , it occurs to me that we should not be sending SQS events based on invoice payment results (I doubt we need to update our cache either). When an invoice fails, that doesn't actually mean they immediately lose access to the product. An invoice can fail for many reasons, and may be retried multiple times before success. When it can no longer be invoice'd, the subscription is cancelled/deleted, and our webhook handler for that event will propagate the subscription state change. (flagging @jackiemunroe to double-check me on this, but I believe this is correct).

Reply:
> Right, I mean, we could possibly avoid 1 stripe call by having a function that can use a uid/email instead to fetch from our Redis customer cache and match the subscription id.

But in this invoice payment hook, we don't have access to a full customer record and so don't have an FxA UID at this point.

Reply:
Oh, I meant that if we fetch the customer from Stripe, we'll then have the uid/email. That will be 1 call (if they're then in Redis), instead of the 2 calls this has right now.

Reply:
@bbangert you are correct. Once Stripe has deemed that an invoice is not collectable, there will be a state change on the invoice and the subscription, both of which will trigger webhooks.

Reply:
But, I believe that the Subscription would move to an "incomplete_expired" state

Reply:
It looks like "incomplete_expired" is only if its the very first payment. Recurring payments will go to "past_due" or be cancelled depending on settings (ours indicate it goes to cancelled).